server/protocols/jmap/src/main/java/org/apache/james/jmap/utils/JsoupHtmlTextExtractor.java

```diff
@@ -0,0 +1,78 @@
+/****************************************************************
+ O * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.utils;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.nodes.Node;
+import org.jsoup.nodes.TextNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JsoupHtmlTextExtractor implements HtmlTextExtractor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JsoupHtmlTextExtractor.class);
+
+    @Override
+    public String toPlainText(String html) {
+        try {
+            Document document = Jsoup.parse(html);
+
+            Element body = Optional.ofNullable(document.body()).orElse(document);
+
+            return flatten(body)
+                .map(this::convertNodeToText)
+                .reduce((s1, s2) -> s1 + s2).orElse("");
```

Comment:
Please prefer the reduce version that accept a identity element instead of the one returning a maybe